uarray/uarray/core.py

```diff
@@ -1,281 +1,280 @@
-# pylint: disable=E1120,W0108,W0621,E1121,E1101
-import matchpy
-import typing
 import functools
+import typing
 from .machinery import *
 
-__all__ = [
-    "Sequence",
-    "Call",
-    "Value",
-    "scalar",
-    "vector",
-    "vector_of",
-    "Unbound",
-    "Scalar",
-    "Content",
-    "GetItem",
-    "Function",
-    "function",
-    "Length",
-    "to_expression",
-    "VectorCallable",
-    "PushVectorCallable",
-    "with_dims",
-    "Unify",
-    "unbound",
-    "VectorIndexed",
-    "gensym",
-    "with_shape",
-    "unbound_with_shape",
-]
-
-
-class Sequence(matchpy.Operation):
-    """
-    Sequence(length, getitem)
-    """
+RET = typing.TypeVar("RET")
+ARG1 = typing.TypeVar("ARG1")
+ARG2 = typing.TypeVar("ARG2")
 
-    name = "Sequence"
-    arity = matchpy.Arity(2, True)
+# Categories
+class CArray:
+    pass
 
 
-class GetItem(matchpy.Operation):
-    name = "GetItem"
-    arity = matchpy.Arity(1, True)
+class CContent:
+    pass
 
 
-register(GetItem(Sequence(w._, w.getitem)), lambda _, getitem: getitem)
+class CUnbound:
+    pass
 
 
-class Length(matchpy.Operation):
-    name = "Length"
-    arity = matchpy.Arity(1, True)
+class CCallableUnary(typing.Generic[RET, ARG1]):
+    pass
 
 
-register(Length(Sequence(w.length, w._)), lambda _, length: length)
+class CCallableBinary(typing.Generic[RET, ARG1, ARG2]):
+    pass
 
 
-class Scalar(matchpy.Operation):
-    """
-    Scalar(content)
-    """
+CGetitem = CCallableUnary[CArray, CContent]
```

Comment:
Is it possible to rename this to `CGetItem`?

Reply:
Yes :) Thanks for catching this.